packages/gatsby/src/redux/persist.ts

```diff
@@ -1,12 +1,155 @@
+import path from "path"
 import v8 from "v8"
-import { readFileSync, writeFileSync } from "fs-extra"
-import { ICachedReduxState } from "./types"
+import {
+  existsSync,
+  mkdtempSync,
+  readFileSync,
+  removeSync,
+  renameSync,
+  writeFileSync,
+} from "fs-extra"
+import { IReduxNode, ICachedReduxState } from "./types"
+import { sync as globSync } from "glob"
 
-const file = (): string => `${process.cwd()}/.cache/redux.state`
+const legacyCacheFile = path.join(process.cwd(), `.cache/redux.state`)
+const reduxCacheFolder = path.join(process.cwd(), `.cache/redux`)
```

Comment:
I remembered why this was a function instead of straight assignment - there was this PR - https://github.com/gatsbyjs/gatsby/pull/19800 I did make a note that it's not guaranteed to continue working in the future, but if there is low effort way to keep theirs usage - that would be awesome

Reply:
I see. I'll add a comment for this as well. Seeing how it's now a functional function (it takes an arg) I think we'll be keeping it a little longer anyways :)